src/view/configuration/dataCollectionSection.jsx

```diff
@@ -223,6 +263,12 @@ const DataCollectionSection = ({ instanceFieldName }) => {
   const [{ value: activityCollectorEnabled }] = useField(
     "components.activityCollector",
   );
+  // Always mark the onBeforeLinkClickSend field to be touched so it will show errors immediately when
+  // event grouping is used.
+  const [, , { setTouched }] = useField(
+    `${instanceFieldName}.onBeforeLinkClickSend`,
+  );
+  useEffect(() => setTouched(true), []);
```

Comment:
You need to have something like this 

```
useEffect(() => { setTouched(true) }, []);
```

Right now you will see this in the log: It looks like you wrote useEffect(async () => ...) or returned a Promise. Instead, write the async function inside your effect and call it immediately: